### ISSUE TITLE
s3key is not idempotent

### ISSUE THREAD

**Issue by simoncolincap** [ionos-cloud/module-ansible#67]
### Current SDK Version

```
Collection               Version
------------------------ -------
ionoscloudsdk.ionoscloud 6.1.2
```

### Use-cases

Currently we have a playbook where we use ansible to create a user, create an S3 key for that user and then use that key for some S3 operations, unfortunately subsequent ansible runs will always create a new pair until we reach the limit of 5 keys per account at which point we get an error and have to manually delete the keys
### Proposal

We need an option to use ansible to ensure that at least one s3 key exists and to retrieve its value, maybe this could be enabled with an optional parameter to keep the default behavior the same.

**Comment by avirtopeanu** [comment]
Hello @simoncolincap, thanks for flagging this issue. We're looking into this and we will keep you updated.

Have a nice day!

**Comment by avirtopeanu-ionos** [comment]
Hey @simoncolincap! [We released a new version, 6.1.3](https://galaxy.ansible.com/ionoscloudsdk/ionoscloud), which enables use of the flag `idempotency`. This will change the behaviour such that an already existing s3key will be returned instead of creating more. By default the behaviour has remained unchanged.

A new module `s3key_info` was also added, which you can use to list all of the s3keys of a specific user. You can view the full changelog here: https://docs.ionos.com/ansible/changelog. There will also be a v5 release soon which will include this feature.

Have a nice day!

**Comment by aimee-889** [comment]
even though it is supposed to be fixed in version 5.0.11 I get the following error: version 5.0.11 Unsupported parameters for (ionoscloudsdk.ionoscloud.s3key) module: idempotency Supported parameters include: active, api_url, key_id, password, state, token, user_id, username, wait, wait_timeout

**Comment by avirtopeanu-ionos** [comment]
Hello @aimee-889, thanks for pointing this out, we will roll out a fix soon

**Comment by avirtopeanu-ionos** [comment]
Hi! This is now fixed in v5.0.12. Have a good day!
